Src/MoneyFox.Windows.Tasks/LiveTilesTaskService.cs

```diff
@@ -0,0 +1,1253 @@
+								{
+									td = td.AddMonths(1);
+								};
+								while (DateTime.Compare(dt, td) <= 0 && i < 8)
+								{
+									p.Data.Date = td;
+									p.Data.ChargedAccount = p.Data.RecurringPayment.ChargedAccount;
+									p.Data.Amount = p.Data.RecurringPayment.Amount;
+									allpayment.Add(td, p);
+									alldates.Add(td);
+									td = td.AddMonths(1);
+									i += 1;
+								};
+								break;
+							}
+						case Foundation.PaymentRecurrence.Bimonthly:
+							{
+								DateTime td = p.Data.RecurringPayment.StartDate;
+								while (DateTime.Compare(dt, td) > 0)
+								{
+									td = td.AddMonths(2);
+								};
+								while (DateTime.Compare(dt, td) <= 0 && i < 8)
+								{
+
+									p.Data.Date = td;
+									p.Data.ChargedAccount = p.Data.RecurringPayment.ChargedAccount;
+									p.Data.Amount = p.Data.RecurringPayment.Amount; 
+									allpayment.Add(td, p);
+									alldates.Add(td);
+									td = td.AddMonths(2);
+								};
+								break;
+							}
+						case Foundation.PaymentRecurrence.Quarterly:
+							{
+								DateTime td = p.Data.RecurringPayment.StartDate;
+								while (DateTime.Compare(dt, td) > 0)
+								{
+									td = td.AddMonths(4);
+								};
+								while (DateTime.Compare(dt, td) <= 0 && i < 8)
+								{
+									p.Data.Date = td;
+									p.Data.ChargedAccount = p.Data.RecurringPayment.ChargedAccount;
+									p.Data.Amount = p.Data.RecurringPayment.Amount;
+									allpayment.Add(td, p);
+									alldates.Add(td);
+									td = td.AddMonths(4);
+									i += 1;
+								};
+								break;
+							}
+						case Foundation.PaymentRecurrence.Biannually:
+							{
+								DateTime td = p.Data.RecurringPayment.StartDate;
+								while (DateTime.Compare(dt, td) > 0)
+								{
+
+									td = td.AddMonths(6);
+								};
+								while (DateTime.Compare(dt, td) <= 0 && i < 8)
+								{
+									p.Data.Date = td;
+									p.Data.ChargedAccount = p.Data.RecurringPayment.ChargedAccount;
+									p.Data.Amount = p.Data.RecurringPayment.Amount;
+									allpayment.Add(td, p);
+									alldates.Add(td);
+									td = td.AddMonths(6);
+									i += 1;
+								};
+								break;
+							}
+						case Foundation.PaymentRecurrence.Yearly:
+							{
+								DateTime td = p.Data.RecurringPayment.StartDate;
+								while (DateTime.Compare(dt, td) > 0)
+								{
+
+									td = td.AddYears(1);
+								};
+								while (DateTime.Compare(dt, td) <= 0 && i < 8)
+								{
+									p.Data.Date = td;
+									p.Data.ChargedAccount = p.Data.RecurringPayment.ChargedAccount;
+									p.Data.Amount = p.Data.RecurringPayment.Amount;
+									allpayment.Add(td, p);
+									alldates.Add(td);
+									td = td.AddYears(1);
+								};
+								break;
+							}
+						default:
+							break;
+					}
+					break;
+				default:
+					break;
+			}
+		}
+		private void GetrecurrancepaymentsMonthly(Payment p, Foundation.PaymentRecurrence reccur, int month)
```

Comment:
Same as a above: I think it would make sense to breka that method a bit up. with close to 300 lines it's quite long. For example you could break out the code in the cases into a own method. Or maybe the strategy pattern might be helpful: https://www.dofactory.com/net/strategy-design-pattern

:)

Reply:
i believe i have made the changes requested